### ISSUE TITLE
arrow-parens and types

### ISSUE THREAD

**Issue by zertosh** [eslint/eslint#7406]
**Tell us about your environment**
- **ESLint Version:** 3.8.1

**What parser (default, Babel-ESLint, etc.) are you using?**

babel-eslint@7.0.0

**Please show your full configuration:**

```
{
  'arrow-parens': [1, 'as-needed'],
}
```

**What did you do? Please include the actual source code causing the issue.**

``` js
 const fn = (foo: FooT) => {}
```

**What did you expect to happen?**

No `arrow-parens` issue.

**What actually happened? Please include the actual, raw output from ESLint.**

`arrow-parens` mistakenly is suggesting the removal of the parens around `foo` - doing so would be a syntax error.

**Comment by platinumazure** [comment]
Thanks for reporting this! I agree this is a bug.
